src/ipfs/uploadIPFSContent.ts

```diff
@@ -0,0 +1,223 @@
+import axios from 'axios';
+
+import { ContentType, IPFSContentAndType, IPFSUploadServiceResponse } from '../types/ipfs';
+
+const ethereumAddressRegex = /^(0x[a-f0-9]{40})$/i;
+
+const uploadToIPFS = async (nexusApiUrl: string, ipfsContentAndType: IPFSContentAndType) => {
+  const [type, content] = ipfsContentAndType;
+  if (!nexusApiUrl) {
+    throw new Error('IPFS_GATEWAY_URL is not set');
+  }
+
+  // POST data to BE service
+  const ipfsHash = await axios
+    .post<IPFSUploadServiceResponse>(nexusApiUrl + '/ipfs', { type, content })
+    .then(response => response.data.ipfsHash)
+    .catch(error => {
+      console.error('Error:', error);
+      throw new Error('Failed to upload data to IPFS');
+    });
+
+  return ipfsHash;
+};
+
+/**
+ *  Uploads data to IPFS
+ * @param {IPFSContentAndType} typeAndContent  - The type of content and the content to be uploaded
+ * @returns {Promise<string>} hash Returns the IPFS hash of the uploaded data
+ *
+ * @example
+ * ```ts
+ * uploadIPFSData(ContentType.coverValidators, { version: '1.0', validators: ['1', '2', '3'] });
+ *
+ * uploadIPFSData(ContentType.coverQuotaShare, { version: '1.0', quotaShare: 25 });
+ *
+ * uploadIPFSData(ContentType.coverAumCoverAmountPercentage, { version: '1.0', aumCoverAmountPercentage: 15 });
+ *
+ * uploadIPFSData(ContentType.coverWalletAddresses, { version: '1.0', walletAddresses: ['0x1', '0x2', '0x3'] });
+ *
+ * uploadIPFSData(ContentType.coverFreeText, { version: '1.0', freeText: 'This is a free text' });
+ * ```
+ */
+export const uploadIPFSContent = async (
+  ipfsContentAndType: IPFSContentAndType,
+  nexusApiUrl: string = 'https://api.nexusmutual.io/v2',
```

Comment:
the other function has no default value for the url, this one does

Reply:
`uploadToIPFS` - is hidden as its the direct upload (no validation). its only called within `uploadIPFSContent` and nowhere else. so its also guaranteed to have the default value from `uploadIPFSContent`